### PR TITLE
Test for pod securityContexts running as non-root

### PR DESCRIPTION
**Pull request by shmanu017**
## Description

Test for pod securityContexts running as non-root

## Related Issues

https://github.com/astronomer/issues/issues/5339

### CODE REVIEW

File: charts/astronomer/templates/astro-ui/astro-ui-deployment.yaml
Comment (danielhoherd) [inline charts/astronomer/templates/astro-ui/astro-ui-deployment.yaml:43]:
```suggestion
          securityContext:{{ toYaml (.Values.securityContext)| nindent 12 }}
```

File: charts/nats/values.yaml
Comment (danielhoherd) [inline charts/nats/values.yaml:16]:
```suggestion

```

File: charts/postgresql/templates/statefulset-slaves.yaml
Comment (danielhoherd) [inline charts/postgresql/templates/statefulset-slaves.yaml:95]:
```suggestion
          runAsNonRoot: {{ .Values.securityContext.runAsNonRoot }}
```

File: charts/prometheus-blackbox-exporter/values.yaml
Comment (danielhoherd) [inline charts/prometheus-blackbox-exporter/values.yaml:14]: Are you adding these blank lines intentionally? I made a suggestion above to remove them, but if you're adding them on purpose then disregard the suggestions.
```suggestion
```

**Review by danielhoherd — COMMENTED** [review]
Looking good so far. I left some optional suggestions, but I think the `prettier` pre-commit hook should take care of them.

File: charts/postgresql/templates/statefulset-slaves.yaml
Comment (danielhoherd) [inline charts/postgresql/templates/statefulset-slaves.yaml:96]: This should be aligned with the `if` block, which I think is:
```suggestion
        {{- end }}
```

File: tests/chart_tests/test_non_root_user.py
Comment (danielhoherd) [inline tests/chart_tests/test_non_root_user.py:16]: Please add a docstring and type annotation here so it's easier to understand.

File: tests/chart_tests/test_non_root_user.py
Comment (danielhoherd) [inline tests/chart_tests/test_non_root_user.py:23]: This might be better understood as a dict comprehension, and would be shorter.

File: tests/chart_tests/test_non_root_user.py
Comment (danielhoherd) [inline tests/chart_tests/test_non_root_user.py:33]: Would it make more sense to have this parametrize on the class definition?

File: tests/chart_tests/test_non_root_user.py
Comment (danielhoherd) [inline tests/chart_tests/test_non_root_user.py:37]: We should probably use some form of `yaml.load` here instead of regex. Doing so would be more specific, more legible, and potentially use fewer resources.

**Review by danielhoherd — APPROVED** [review]
LGTM

**Comment by shmanu017** [comment]
Changes are working as expecting on a Test cluster, need to update image tags before merging to master.

File: charts/alertmanager/values.yaml
Comment (danielhoherd) [inline charts/alertmanager/values.yaml:17]: Is this needed? Does the docker image already run as 65534?

File: charts/astronomer/values.yaml
Comment (danielhoherd) [inline charts/astronomer/values.yaml:18]: This will need to be reverted.

File: charts/astronomer/values.yaml
Comment (danielhoherd) [inline charts/astronomer/values.yaml]: This will also need to be reverted.

File: charts/kube-state/values.yaml
Comment (danielhoherd) [inline charts/kube-state/values.yaml:17]: This needs a value.

**Comment by shmanu017** [comment]
`commander` , `houston-api` and `db-bootstraper` are currently pointing to master or main , as it will either way fail when running  test on cluster enlivenment.
Waiting on merge the change until the release tag are released.

File: charts/astronomer/values.yaml
Comment (danielhoherd) [inline charts/astronomer/values.yaml:61]: I assume this deletion was a mistake
```suggestion
  # This only applies when replicas > 3
```

**Review by danielhoherd — APPROVED** [review]
LGTM. I like the two additions of PodSecurityContext if those are the only two places that differentiation is needed.

**Pull request merged**
